src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesVsPatterns.cs

```diff
@@ -0,0 +1,1336 @@
+                //         y4.ToString(); // 7
+                Diagnostic(ErrorCode.ERR_UseDefViolation, "y4").WithArguments("y4").WithLocation(27, 9));
+        }
+
+        [Fact]
+        public void UnconstrainedTypeParameter_PatternMatching()
+        {
+            var source =
+@"
+class C
+{
+    static void F1<T>(object o, T tin)
+    {
+        if (o is T t1)
+        {
+            t1.ToString();
+        }
+        else
+        {
+            t1 = default; // 1
+        }
+
+        t1.ToString(); // 2
+
+        if (!(o is T t2))
+        {
+            t2 = tin;
+        }
+        else
+        {
+            t2.ToString();
+        }
+        t2.ToString(); // 3
+
+        if (!(o is T t3)) return;
+        t3.ToString();
+    }
+}
+";
+
+            var comp = CreateCompilation(new[] { source }, options: WithNonNullTypesTrue());
+            comp.VerifyDiagnostics(
+                // (12,18): warning CS8652: A default expression introduces a null value when 'T' is a non-nullable reference type.
+                //             t1 = default; // 1
+                Diagnostic(ErrorCode.WRN_DefaultExpressionMayIntroduceNullT, "default").WithArguments("T").WithLocation(12, 18),
+                // (15,9): warning CS8602: Possible dereference of a null reference.
+                //         t1.ToString(); // 2
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "t1").WithLocation(15, 9),
+                // (25,9): warning CS8602: Possible dereference of a null reference.
+                //         t2.ToString(); // 3
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "t2").WithLocation(25, 9)
+                );
+        }
+
+        [WorkItem(32503, "https://github.com/dotnet/roslyn/issues/32503")]
+        [Fact]
+        public void PatternDeclarationBreaksNullableAnalysis()
+        {
+            var source = @"
+#nullable enable
+class A { }
+class B : A
+{
+    A M()
+    {
+        var s = new A();
+        if (s is B b) {}
+        return s; 
+    }
+} 
+";
+            var comp = CreateCompilation(source);
+            comp.VerifyDiagnostics();
+        }
+
+        [Fact]
+        public void InferenceWithITuplePattern()
+        {
+            var source = @"
+#nullable enable
+class A { }
+class B : A
+{
+    A M()
+    {
+        var s = new A();
+        if (s is B b) {}
+        return s; 
+    }
+} 
+";
+            var comp = CreateCompilation(source);
+            comp.VerifyDiagnostics();
+        }
+
+        [Fact]
+        public void RecursivePatternNullInferenceWithDowncast_01()
+        {
+            var source = @"
+#nullable enable
+class Base
+{
+    public object Value = """";
+}
+class Derived : Base
+{
+    public new object Value = """";
+}
+class Program
+{
+    void M(Base? b)
+    {
+        if (b is Derived { Value: null })
+            b.Value.ToString();
+    }
+}";
+            var comp = CreateCompilation(source);
+            comp.VerifyDiagnostics();
```

Comment:
Is the fact that there are no diagnostics here tracked?

Reply:
Oh, it's because the `Value` being checked is not the `Value` being accessed.

---
In reply to: [267958632](https://github.com/dotnet/roslyn/pull/34273#discussion_r267958632) [](ancestors = 267958632)